src/MainWindow.vala

```diff
@@ -443,6 +443,14 @@ namespace PantheonCalculator {
                 case Gdk.Key.Escape:
                     button_clr_clicked ();
                     break;
+                case Gdk.Key.KP_Decimal:
+                case Gdk.Key.KP_Separator:
+                case Gdk.Key.decimalpoint:
+                case Gdk.Key.period:
+                case Gdk.Key.comma:
+                    decimal () ;
```

Comment:
There's no need to separate two lines of code into a new function. Just put everything from `decimal` here.

Reply:
That was the solution the Gnome team used for their calculator, I've been using it for 2 months now and it prevented me to get my HP 12C from my draw every time I wanted to calc something. =D
I'll make the changes you've asked! hope it's better now!